receiver/splunkhecreceiver/receiver.go

```diff
@@ -161,7 +194,9 @@ func (r *splunkReceiver) handleReq(resp http.ResponseWriter, req *http.Request)
 		transport = "https"
 	}
 	ctx := obsreport.ReceiverContext(req.Context(), r.config.Name(), transport, r.config.Name())
-	ctx = obsreport.StartMetricsReceiveOp(ctx, r.config.Name(), transport)
+	if r.logsConsumer == nil {
+		ctx = obsreport.StartMetricsReceiveOp(ctx, r.config.Name(), transport)
+	}
```

Comment:
I am not sure I understand what this does. I'd expect this to check `if metricsConsume!=nil` if I understand the intent correctly.

Which brings to the question, where is our corresponding `StartLogsReceiveOp`? We probably need to add it (can be a future PR).

Reply:
Right now the receiver only exists if the logsConsumer is nil or the metricsConsumer is nil. You can safely assume _for now_ that if logsConsumer is nil, the metricsConsumer is not. This `if` is probably misleading though, sorry. I can make it check if r.metricsConsumer is not nil instead.

And yes, there is no `StartLogsReceiveOp` that I could find at this time.

That said, read on - there are a number of subtleties about HEC input.